src/txn/txn_ckpt.c

```diff
@@ -140,10 +141,22 @@ __checkpoint_flush_tier(WT_SESSION_IMPL *session, bool force)
             WT_ERR(__wt_session_get_dhandle(session, key, NULL, NULL, 0));
             /*
              * When we call wt_tiered_switch the session->dhandle points to the tiered: entry and
-             * the arg is the config string that is currently in the metadata.
+             * the arg is the config string that is currently in the metadata. Also, ensure that
+             * this table participates in the checkpoint process, even if clean.
              */
             WT_ERR(__wt_tiered_switch(session, value));
             WT_STAT_CONN_INCR(session, flush_tier_switched);
+            btree = S2BT(session);
+            WT_BTREE_CLEAN_CKPT(session, btree, WT_BTREE_CLEAN_CKPT_NOW);
```

Comment:
I will check if I can just mark this tree modified and hence it gets checkpointed.